umbraco/Umbraco-CMS
# Package format changed to zip

This changes the package format back to a zip instead of just an xml file. Why? Because this is the only good way we can support media. The Zip will consist of a package.xml at the root and then the media files stored in a /media root folder of the zip and the paths will be the paths returned by the media file system provider. By default we will end up with paths like /media/media/asdfasdf/blah.jpg because the default provider stores in the media root path. This is ok, we just put the additional /media root folder in the zip to ensure that if we want to add other assets to the zip we can in differently named root folders.

* During serialization/packaging of media:
  * We collect all media file paths - based on their current property editor - and store those as part of the serialized xml.
  * During serialization we collect a list of streams/files to include in the zip for media.
  * Now we have serialized media as XML with their corresponding files and paths which are included in the zip which can be embedded.
* During package migration:
  * We need to first install all media as items because we need this data to then re-save with the uploaded media item
  * We then correlate the media item and the media file included in the zip and then call media.SetValue with the stream which will re-set the value for the media item to the correct path based on the current media file system provider  

This also fixes the hacks where in several places the media path was extracted by checking for json values and then just assuming that it's the image cropper and/or having a "src" parameter which is incorrect. We have ways of getting the media path based on the MediaUrlGeneratorCollection/IMediaUrlGenerator.

This removes OriginalMediaPathScheme, it is obsolete, buggy and it shouldn't be used and it requires more complicated APIs and overloads which I've also now removed.

## Testing

* Create some media in a folder, images, etc... 
* Create a package with that media in it (including descendants)
* Save and download the package
* Verify the zip downloaded, that it has the files in the right place according to the above description
* Go delete your media, and empty recycle bin
* Create a new auto package migration plan like:
  ```cs
   public class TestAutomaticPackageMigrationPlan : AutomaticPackageMigrationPlan
   {
       public TestAutomaticPackageMigrationPlan()
           : base("TestZipPackage")
       {
       }
   }
  ```
* embed the package.zip file in the same namespace
* stop your site
* start your site
* the package will automatically be installed including your media items and your media files, verify this in the media section
